app/models/school.rb

```diff
@@ -48,16 +48,14 @@ def religious_character
   private
 
   def set_geolocation_from_easting_and_northing
-    if easting && northing
-      wgs84 = Breasal::EastingNorthing.new(
-        easting: easting.to_i,
-        northing: northing.to_i,
-        type: :gb,
-      ).to_wgs84
-
-      geolocation = [wgs84[:latitude], wgs84[:longitude]]
-    end
-
-    self.geolocation = geolocation
+    self.geolocation = if easting && northing && (easting_changed? || northing_changed?)
```

Comment:
a guard clause would be more readable here:

https://rubystyle.guide/#no-nested-conditionals

Reply:
I tried a guard clause but actually it changes the behaviour since we want to set geolocation to nil if the condition is false.